gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsInfoApiMonitor.java

```diff
@@ -47,18 +63,42 @@ public ClusterStats monitor(ProxyBackendConfiguration backend)
     }
 
     private boolean isReadyStatus(String baseUrl)
+    {
+        return isReadyStatus(baseUrl, retries);
+    }
+
+    private boolean isReadyStatus(String baseUrl, int retriesRemaining)
     {
         Request request = prepareGet()
                 .setUri(uriBuilderFrom(URI.create(baseUrl)).appendPath("/v1/info").build())
                 .build();
-
         try {
             ServerInfo serverInfo = client.execute(request, SERVER_INFO_JSON_RESPONSE_HANDLER);
             return !serverInfo.isStarting();
         }
+        catch (UnexpectedResponseException e) {
+            if (shouldRetry(e.getStatusCode())) {
+                if (retriesRemaining > 0) {
+                    return isReadyStatus(baseUrl, retriesRemaining - 1);
+                }
+            }
+            log.warn(e, "Health check failed with non-retryable response. %s", e.toString());
```

Comment:
Not only does non-retryable response get here. We will also end up in this statement if `retries == 0`. We should add something to distinguish between these two states

Reply:
Why is this a warning and not an error? even though it's non retry-able response.